src/openfl/_internal/renderer/opengl/VertexArrayObjectUtils.hx

```diff
@@ -0,0 +1,114 @@
+package openfl._internal.renderer.opengl;
+
+
+import lime.graphics.GLRenderContext;
+import lime.graphics.opengl.GLContextType;
+import lime.graphics.opengl.GLVertexArrayObject;
+
+
+/**
+*  VertexArrayObjectUtils is a utility class providing a utility methods to work with VertextArrayObjects.
+**/
+class VertexArrayObjectUtils {
+	
+	
+	private static var __initialized:Bool = false;
+	private static var __enabled:Bool = true;
+	private static var __vertexArrayObjectsSupported:Bool = false;
+	private static var __vertexArrayObjectsExtension:Dynamic = null;
+	
+	
+	public static inline function bindVAO (gl:GLRenderContext, vao:GLVertexArrayObject):Void {
+		
+		#if vertex_array_object
+		if (!isVertexArrayObjectsSupported(gl)) return;
+		
+		if (__vertexArrayObjectsExtension != null) {
+			
+			__vertexArrayObjectsExtension.bindVertexArrayOES (vao);
+			
+		} else {
+			
+			gl.bindVertexArray (vao);
+			
+		}
+		#end
+		
+	}
+	
+	
+	public static function createVAO (gl:GLRenderContext):GLVertexArrayObject {
+		
+		#if vertex_array_object
+		if (!isVertexArrayObjectsSupported(gl)) return null;
+		
+		if (__vertexArrayObjectsExtension != null) {
+			
+			return __vertexArrayObjectsExtension.createVertexArrayOES ();
+			
+		} else {
+			
+			return gl.createVertexArray ();
+			
+		}
+		#end
+		
+		return null;
+		
+	}
+	
+	
+	public static inline function deleteVAO (gl:GLRenderContext, vao:GLVertexArrayObject):Void {
+		
+		#if vertex_array_object
+		if (!isVertexArrayObjectsSupported(gl) || vao == null) return;
+		
+		if (__vertexArrayObjectsExtension != null) {
+			
+			__vertexArrayObjectsExtension.deleteVertexArrayOES (vao);
+			
+		} else {
+			
+			gl.deleteVertexArray (vao);
+			
+		}
+		#end
+		
+	}
+	
+	
+	private static inline function __init (gl:GLRenderContext):Void {
+		
+		__initialized = true;
+		#if vertex_array_object
+		if (gl.type == GLContextType.WEBGL) { 
+			
+			if (gl.version == 2) {
+				
+				__vertexArrayObjectsSupported = true;
+				
+			} else if (gl.version == 1) {
+				
+				__vertexArrayObjectsExtension = gl.getExtension ("OES_vertex_array_object");
+				__vertexArrayObjectsSupported = __vertexArrayObjectsExtension != null;
+				
+			}
+			
+		}
+		#end
+		
+	}
+	
+	
+	public static inline function isVertexArrayObjectsSupported(gl:GLRenderContext):Bool {
+		#if vertex_array_object
+		if (!__enabled) return false;
+		if (!__initialized) __init (gl);
+		#end
```

Comment:
I was thinking long on this. First implementation was with public init method called from GLRenderer. To be honest not sure what is the best solution. If we have public init method I'd need to call it from Starling as well, since it is possible that non-webgl rendering is used for UI in combination with Starling, so the GLRenderer won't be created and therefore VertexArrayObjectUtils won't be initialized in openfl. 
I don't like as well doing these checks 
```
if (!__enabled) return false;
if (!__initialized) __init (gl);
```
every time you access VertexArrayObjectUtils methods, why I like its robustness, meaning that its methods will always work. Even if you forget to initialize it. 
I think about it and ask other guys for opinions so we decide together what's best.

Reply:
Yes, so my instinct went into the right direction ;) I would prefer a public init, but I totally get your point